core/src/main/scala/org/bykn/bosatsu/DefStatement.scala

```diff
@@ -26,7 +27,13 @@ case class DefStatement[A, B](
       }
 
     (args.traverse(argFn), bodyExp, tag).mapN { (as, b, t) =>
-      Expr.buildPatternLambda(as, b, t)
+      val lambda = Expr.buildPatternLambda(as, b, t)
+      typeArgs match {
+        case None => lambda
+        case Some(args) =>
+          val bs = args.map { case TypeRef.TypeVar(b) => rankn.Type.Var.Bound(b) }
+          Expr.Generic(bs, lambda)
```

Comment:
We need to check in SourceConverter, which may be the only caller of this method, that all the declared type parameters are actually free.

We shouldn't allow totally free parameters. (In general, declaring forall x. y should require that x is free in y, otherwise it is likely a typo).